api/src/main/java/org/apache/iceberg/variants/VariantDataUtil.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.variants;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.base.Splitter;
+
+public class VariantDataUtil {
+  private VariantDataUtil() {}
+
+  private static final String RFC9535_NAME_FIRST =
+      "[A-Za-z_\\x{0080}-\\x{D7FF}\\x{E000}-\\x{10FFFF}]";
+  private static final String RFC9535_NAME_CHARS =
+      "[0-9A-Za-z_\\x{0080}-\\x{D7FF}\\x{E000}-\\x{10FFFF}]*";
+  private static final Predicate<String> RFC9535_MEMBER_NAME_SHORTHAND =
+      Pattern.compile(RFC9535_NAME_FIRST + RFC9535_NAME_CHARS).asMatchPredicate();
+
+  private static final Splitter DOT = Splitter.on(".");
+  private static final String ROOT = "$";
+
+  public static List<String> parsePath(String path) {
```

Comment:
I don't like that we're using a `List<String>` to handle this as I feel it's going to get much more complicated over time.  Why not put this in a wrapper of `VariantPath`?  I think it would be better not to leak this through a public interface.

Reply:
In fact, I think this whole class should just be redefined as `VarientPath` since that's all it's really doing at this point and then we can build on this going forward.

Reply:
I'll move this and make it package-private so we don't need to worry about changing it later.

Reply:
This is now package-private so we can modify it later, but my follow-up PR should still work.